src/visit.rs

```diff
@@ -1122,6 +1130,9 @@ pub fn visit_declarator<'ast, V: Visit<'ast> + ?Sized>(
     _span: &'ast Span,
 ) {
     visitor.visit_declarator_kind(&declarator.kind.node, &declarator.kind.span);
+    for pointer in &declarator.pointer {
```

Comment:
Maybe it makes sense here to visit pointer declarators in reverse order? This way all declarators can be applyed to the base type consistently.